packages/react-native/ReactCommon/react/performance/timeline/CircularBuffer.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <functional>
+#include <vector>
+#include "PerformanceEntry.h"
+
+namespace facebook::react {
+
+constexpr size_t DEFAULT_MAX_SIZE = 1024;
+
+/**
+ * A container for storing entries of type T, with the following properties:
+ * - It can only grow up to a specified max size
+ * - It's a circular buffer (the oldest elements are dropped if reached max
+ * size and adding a new element)
+ *
+ * Note that the space for maxSize elements is reserved on construction. This
+ * ensures that pointers to elements remain stable across add() operations.
+ */
+template <class T>
+class CircularBuffer {
+ public:
+  explicit CircularBuffer(size_t maxSize = DEFAULT_MAX_SIZE)
```

Comment:
Nit: I'd define a default for something like initial size, but for something that will define when you're going to start dropping data, it's probably better to force the user to specify a value.